src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -855,7 +855,35 @@ void InferRequest::prepare_input(const cldnn::primitive_id& inputName, Blob::Ptr
     };
 
     auto _nw_ptr = m_graph->GetNetwork();
-    if (input_layout.is_dynamic()) {
```

Comment:
[random spot]
It seems that we should limit the usage of this optimization to scenarios involving single stream and single infer request only, isn't it?

Reply:
hmm why so?

Reply:
Ah yes, you are right. I will update with fix! thanks!